### DISCUSSION TITLE
Client unable to signin/signup with Anon Key

### DISCUSSION THREAD

**Discussion by rancharles** [supabase/supabase#9177] [Questions]
I'm setting up client-side authentication in SvelteKit with supabase-js. For some reason, the user is unable to signin/signup with anon key. I get error an 400 with 'request disallowed'. I'm very certain it's not my code logic at fault, since I can signin/signup fine when I use the service key instead of anon key (tested locally) when initializing supabaseClient, and changing literally nothing else. I'm also sure that I did copy the anon key correctly. I just followed the sveltekit quickstart guide, and ran the user management starter query. This is my first project using supabase and I feel like I might be missing something trivial but unsure what. Please help!

**Comment by GaryAustin1** [comment]
Can you show your client init and the auth call you are making?

**Reply by rancharles** [reply]
```typescript
import { createClient } from '@supabase/supabase-js';

const supabaseUrl = import.meta.env.VITE_SUPABASE_URL;
const supabaseAnonKey = import.meta.env.VITE_SUPABASE_ANON_KEY;

export const supabase = createClient(supabaseUrl, supabaseAnonKey);
```
```typescript
const handleLogin = async () => {
    try {
        loading = true;
        const { error } = await supabase.auth.signIn({ email: loginEmail, password: loginPassword });
        if (error) throw error;
    } catch ({ error_description, message }) {
        console.log(error_description || message);
    } finally {
        loading = false;
    }
    
    user.set(!!supabase.auth.user());
};
```
I receive error code `400` "request disallowed"
Now, things work perfectly if I just use `VITE_SUPABASE_SERVICE_KEY` instead of `VITE_SUPABASE_ANON_KEY` (I have both inside `.env`)

I'm fairly certain this has to do with some configuration mistake I might've made in the supabase UI. I don't remember doing anything that could warrant this though, but because things work with the service key I doubt my code logic is at fault. I made a new supabase project in which everything works and I'll carefully reconfigure everything again there. But I'm curious what exactly was the problem here, because I have no clue even after exhausting all the possibilities I can think of.

**Reply by GaryAustin1** [reply]
Ok that was going to be my next question,  you did set service_role same way thru VITE var.

Auth/gotrue signUp/signIn should not care which you use as long as they are valid.  

You could look in API logs in the dashboard.
You could do a test and not sign in but just do a select on a non-RLS enabled table to verify anon key is working.
If you have a trigger function on auth.users (like for a profile table) you could remove that, but I would expect a 500 error if it messed up.
You can look at the network tab (if this is not serverside) and see if the apikey and authorization header are set with the jwt you expect.
